### ISSUE TITLE
Create netcdf_meta.sh in parallel to netcdf_meta.h

### ISSUE THREAD

**Issue by DennisHeimbigner** [Unidata/netcdf-c#1850]
A number of times I have needed to get feature information for shell scripts
such as is HDF5 enabled. We have that information for use by programs
using netcdf_meta.h. It might be nice to have an equivalent netcdf_meta.sh
that provides this info for shell scripts.
It would be pretty easy, and in fact with a little reformatting,
we can transform netcdf_meta.h.in to netcdf_meta.sh.in

**Comment by edwardhartnett** [comment]
I was thinking about this just today! ;-)

But then I thought that the scripts are already created from run_test.sh.in scripts by configure. So using AC_SUBST it should already be easy to do that.

**Comment by WardF** [comment]
Would it make sense to add this sort of extended information into nc-config.sh[.in]?

**Comment by edwardhartnett** [comment]
I think adding it to nc-config is a great idea and probably something we should have been doing every time.

**Comment by DennisHeimbigner** [comment]
The only thing is to make sure that our shell script test cases can easily find nc-config.sh
even when doing out-of-tree builds.

**Comment by DennisHeimbigner** [comment]
Point of clarification. Is the proposal to put this in nc-config or to create
a new nc-config.sh for it?

**Comment by edwardhartnett** [comment]
I don't believe we need an nc-config.sh. Instead, do whatever substitutions you need in the run_test.sh.in scripts directly, since variable substitution is already easily available as the shell scripts are created.

In terms of nc-config, I thought that Ward's idea was that all the info we are putting into the build summary and netcdf_meta.h should possibly also be added to nc-config, and I think that's correct. So a bunch of fields should be added to nc-config to reflect whatever is not there now, and then, as we add new definitions to netcdf_meta.h, we should consider each one for inclusion in nc-config as well.

**Comment by WardF** [comment]
Sorry for radio silence as I’m on the road to Oregon.  Hello from Wyoming. Ed is correct in his summary of what I intended by my comment. Thanks all!
